nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1322 @@
+        else:
+            self.nvertices = {self.to_cifti_brain_structure_name(name): number
+                              for name, number in nvertices.items()}
+
+        for name in list(self.nvertices.keys()):
+            if name not in self.name:
+                del self.nvertices[name]
+
+        is_surface = self.is_surface
+        if is_surface.all():
+            self.affine = None
+            self.volume_shape = None
+        else:
+            if affine is None or volume_shape is None:
+                raise ValueError("Affine and volume shape should be defined "
+                                 "for BrainModel containing voxels")
+            self.affine = affine
+            self.volume_shape = volume_shape
+
+        if np.any(self.vertex[is_surface] < 0):
+            raise ValueError('Undefined vertex indices found for surface elements')
+        if np.any(self.voxel[~is_surface] < 0):
+            raise ValueError('Undefined voxel indices found for volumetric elements')
+
+        for check_name in ('name', 'voxel', 'vertex'):
+            shape = (self.size, 3) if check_name == 'voxel' else (self.size, )
+            if getattr(self, check_name).shape != shape:
+                raise ValueError("Input {} has incorrect shape ({}) for BrainModel axis".format(
+                        check_name, getattr(self, check_name).shape))
+
+    @classmethod
+    def from_mask(cls, mask, name='other', affine=None):
+        """
+        Creates a new BrainModel axis describing the provided mask
+
+        Parameters
+        ----------
+        mask : np.ndarray
+            all non-zero voxels will be included in the BrainModel axis
+            should be (Nx, Ny, Nz) array for volume mask or (Nvertex, ) array for surface mask
+        name : str
+            Name of the brain structure (e.g. 'CortexRight', 'thalamus_left' or 'brain_stem')
+        affine : np.ndarray
+            (4, 4) array with the voxel to mm transformation (defaults to identity matrix)
+            Argument will be ignored for surface masks
+
+        Returns
+        -------
+        BrainModel which covers the provided mask
+        """
+        if affine is None:
+            affine = np.eye(4)
+        else:
+            affine = np.asanyarray(affine)
+        if affine.shape != (4, 4):
+            raise ValueError("Affine transformation should be a 4x4 array or None, not %r" % affine)
+        if mask.ndim == 1:
+            return cls.from_surface(np.where(mask != 0)[0], mask.size, name=name)
+        elif mask.ndim == 3:
+            voxels = np.array(np.where(mask != 0)).T
+            return cls(name, voxel=voxels, affine=affine, volume_shape=mask.shape)
+        else:
+            raise ValueError("Mask should be either 1-dimensional (for surfaces) or "
+                             "3-dimensional (for volumes), not %i-dimensional" % mask.ndim)
+
+    @classmethod
+    def from_surface(cls, vertices, nvertex, name='Cortex'):
```

Comment:
if it's left as a default, perhaps we use `Unknown`. i don't think the library should make this choice.

Reply:
I've set it to Other (which is the closest equivalant to Unknown allowed by the CIFTI spec)